[inductor] Simplified TritonKernel.indexing

fdrocha opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #89593
* #89566

Following up on #89566, I tried removing some of the logic in `TritonKernel.indexing` that seems that is no longer necessary. Probably not everything removed in first commit _should_ be removed, as I am not sure what was the intent of some parts, but at least _some_ is no longer necessary with new logic. FWIW, tests seem to pass after removal.


cc @mlazos @soumith @voznesenskym @yanboliang @penguinwu @anijain2305 @EikanWang @jgong5 @Guobing-Chen @chunyuan-w @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @peterbell10 @desertfire

Commit:
[inductor] Simplified TritonKernel.indexing

[ghstack-poisoned]

Commit:
Update on "[inductor] Simplified TritonKernel.indexing"


Following up on #89566, I tried removing some of the logic in `TritonKernel.indexing` that seems that is no longer necessary. Probably not everything removed in first commit _should_ be removed, as I am not sure what was the intent of some parts, but at least _some_ is no longer necessary with new logic. FWIW, tests seem to pass after removal.


cc mlazos soumith voznesenskym yanboliang penguinwu anijain2305 EikanWang jgong5 Guobing-Chen chunyuan-w XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 desertfire

[ghstack-poisoned]

Commit:
Update on "[inductor] Simplified TritonKernel.indexing"


Following up on #89566, I tried removing some of the logic in `TritonKernel.indexing` that seems that is no longer necessary. Probably not everything removed in first commit _should_ be removed, as I am not sure what was the intent of some parts, but at least _some_ is no longer necessary with new logic. FWIW, tests seem to pass after removal.


cc mlazos soumith voznesenskym yanboliang penguinwu anijain2305 EikanWang jgong5 Guobing-Chen chunyuan-w XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 desertfire

[ghstack-poisoned]

jansel reviewed (changes requested):
Test failures seem real.

Commit:
Update on "[inductor] Simplified TritonKernel.indexing"


Following up on #89566, I tried removing some of the logic in `TritonKernel.indexing` that seems that is no longer necessary. Probably not everything removed in first commit _should_ be removed, as I am not sure what was the intent of some parts, but at least _some_ is no longer necessary with new logic. FWIW, tests seem to pass after removal.


cc mlazos soumith voznesenskym yanboliang penguinwu anijain2305 EikanWang jgong5 Guobing-Chen chunyuan-w XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 desertfire

[ghstack-poisoned]

Commit:
Update on "[inductor] Simplified TritonKernel.indexing"


Following up on #89566, I tried removing some of the logic in `TritonKernel.indexing` that seems that is no longer necessary. Probably not everything removed in first commit _should_ be removed, as I am not sure what was the intent of some parts, but at least _some_ is no longer necessary with new logic. FWIW, tests seem to pass after removal.


cc mlazos soumith voznesenskym yanboliang penguinwu anijain2305 EikanWang jgong5 Guobing-Chen chunyuan-w XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 desertfire

[ghstack-poisoned]

Commit:
Update on "[inductor] Simplified TritonKernel.indexing"


Following up on #89566, I tried removing some of the logic in `TritonKernel.indexing` that seems that is no longer necessary. Probably not everything removed in first commit _should_ be removed, as I am not sure what was the intent of some parts, but at least _some_ is no longer necessary with new logic. FWIW, tests seem to pass after removal.


cc mlazos soumith voznesenskym yanboliang penguinwu anijain2305 EikanWang jgong5 Guobing-Chen chunyuan-w XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 desertfire

[ghstack-poisoned]

fdrocha commented:
Closing this, will look into splitting into multiple PRs
